### PR TITLE
Feature load more

### PR DESCRIPTION
**Pull request by ngochieu1297**
## [Pull request title]
Feature load more
### [Redmine link]
https://edu-redmine.framgia.vn/issues/7529

### CODE REVIEW

**Comment by ngochieu1297** [comment]
ready

File: app/src/main/java/com/example/music_58/ui/genre_detail/GenreDetailFragment.java [hunk 1]
```diff
@@ -52,21 +59,60 @@ public static GenreDetailFragment getInstance(Genre genre) {
...
+            String api = StringUtils.initGenreApi(mGenreKey, mOffset);
+            mPresenter.getTracks(api);
+        }
+    }
 
+    @Override
+    public void loadMoreData() {
+        mIsScrolling = false;
+        mProgressBar.setVisibility(View.VISIBLE);
+        mOffset += 20;
```
Comment (NguyenMinhTuanB) [inline app/src/main/java/com/example/music_58/ui/genre_detail/GenreDetailFragment.java]: hardcode

File: app/src/main/res/drawable/custom_item_genres_background.xml [hunk 2]
```diff
@@ -0,0 +1,6 @@
+<?xml version="1.0" encoding="utf-8"?>
+<shape xmlns:android="http://schemas.android.com/apk/res/android"
+    android:shape="rectangle">
+    <corners android:radius="8dp" />
+    <solid android:color="@android:color/white"/>
+</shape>
```
Comment (NguyenMinhTuanB) [inline app/src/main/res/drawable/custom_item_genres_background.xml]: need an endline

File: app/src/main/res/layout/fragment_genre_detail.xml [hunk 3]
```diff
@@ -30,7 +31,7 @@
             <android.support.v7.widget.Toolbar
                 android:id="@+id/toolbar"
                 android:layout_width="match_parent"
-                android:layout_height="?attr/actionBarSize"
+                android:layout_height="100dp"
```
Comment (NguyenMinhTuanB) [inline app/src/main/res/layout/fragment_genre_detail.xml]: hardcode

File: app/src/main/java/com/example/music_58/ui/LoadMoreAbstract.java [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+import android.support.v7.widget.LinearLayoutManager;
+import android.support.v7.widget.RecyclerView;
+import android.widget.AbsListView;
+import android.widget.ProgressBar;
+
+public abstract class LoadMoreAbstract extends Fragment {
+    protected RecyclerView mRecyclerView;
+    protected LinearLayoutManager mLinearLayoutManager;
+    protected boolean mIsScrolling = false;
+    protected int mCurrentItem, mTotalItem, mScrollOutItem;
```
Comment (NguyenMinhTuanB) [inline app/src/main/java/com/example/music_58/ui/LoadMoreAbstract.java]: mỗi biến 1 dòng

**Comment by ngochieu1297** [comment]
ready

File: app/src/main/java/com/example/music_58/util/Constants.java [hunk 5]
```diff
@@ -14,7 +14,7 @@
     public static final int INDEX_UNIT = 1;
     public static final int NEXT_SONG = 1;
     public static final int PREVIOUS_SONG = -1;
-    public static final int LIMIT = 10;
+    public static final int LIMIT = 20;
```
Comment (NguyenMinhTuanB) [inline app/src/main/java/com/example/music_58/util/Constants.java]:
```suggestion
    public static final int SONG_LIMIT = 20;
```

File: app/src/main/java/com/example/music_58/util/Constants.java [hunk 6]
```diff
@@ -14,7 +14,7 @@
     public static final int INDEX_UNIT = 1;
```
Comment (NguyenMinhTuanB) [inline app/src/main/java/com/example/music_58/util/Constants.java]: 14 15 có lặp k?

**Comment by ngochieu1297** [comment]
ready

**Pull request merged**